Sources/AsyncHTTPClient/ConnectionPool/HTTPConnectionPool.swift

```diff
@@ -123,6 +123,46 @@ enum HTTPConnectionPool {
     }
 }
 
+extension HTTPConnectionPool {
+    /// This is a wrapper that we use inside the connection pool state machine to ensure that
+    /// the actual request can not be accessed at any time. Further it exposes all that is needed within
+    /// the state machine. A request ID and the `EventLoop` requirement.
+    struct Request {
+        struct ID: Hashable {
+            let objectIdentifier: ObjectIdentifier
+            let eventLoopID: EventLoopID?
+
+            fileprivate init(_ request: HTTPSchedulableRequest, eventLoopRequirement eventLoopID: EventLoopID?) {
+                self.objectIdentifier = ObjectIdentifier(request)
+                self.eventLoopID = eventLoopID
+            }
+        }
+
+        fileprivate let req: HTTPSchedulableRequest
+
+        init(_ request: HTTPSchedulableRequest) {
+            self.req = request
+        }
+
+        var id: HTTPConnectionPool.Request.ID {
+            HTTPConnectionPool.Request.ID(self.req, eventLoopRequirement: self.requiredEventLoop?.id)
+        }
+
+        var requiredEventLoop: EventLoop? {
```

Comment:
Perhaps I'm misremembering but I think I've seen this same `switch` in a bunch of places; would an extension on whatever the type of `eventLoopPreference` is be worth having?

Reply:
I hope to solve all those places, with this single place. It is really only important when queuing the request. So I hope this is the place!